rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -96,6 +97,31 @@ public NodeMessageHandler(RskSystemProperties config,
             @Nullable TransactionGateway transactionGateway,
             @Nullable PeerScoringManager peerScoringManager,
             StatusResolver statusResolver) {
+        this(
+                config,
+                blockProcessor,
+                syncProcessor,
+                snapshotProcessor,
+                channelManager,
+                transactionGateway,
+                peerScoringManager,
+                statusResolver,
+                null,
+                null
+        );
+    }
+
+    @VisibleForTesting
```

Comment:
[@VisibleForTesting](https://medium.com/@nachare.reena8/exploring-visiblefortesting-for-enhanced-unit-testing-in-android-b45a587ef32f) lets us set up a special constructor for testing. This way, we can pass the parameters we need or mock them for testing purposes. Outside of testing, we want to prevent developers from passing the _Thread_ and _MessageCounter_ when creating an instance of the class